From 3bcb2659dfece96a0cfb22b9b5b7a13485fa863a Mon Sep 17 00:00:00 2001
From: David Vrabel <david.vrabel@citrix.com>
Date: Mon, 14 Mar 2016 17:15:27 +0000
Subject: [PATCH] hvmloader: add high memory e820 region if needed

If the MMIO hole is large and hvmloader needs to relocate memory to
immediately above the 4 GiB boundary, the e820 presented to the guest
will not have a RAM region above 4 GiB.

e.g., a guest with 3 GiB of memory and a 2 GiB MMIO hole will only see
2 GiB.

The required e820 memory region above 4 GiB needs to be added, and not
just filled in.

Signed-off-by: David Vrabel <david.vrabel@citrix.com>
---
 tools/firmware/hvmloader/e820.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/firmware/hvmloader/e820.c b/tools/firmware/hvmloader/e820.c
index bbde2be..5541b18 100644
--- a/tools/firmware/hvmloader/e820.c
+++ b/tools/firmware/hvmloader/e820.c
@@ -99,6 +99,7 @@ void adjust_memory_map(void)
                 ((uint64_t)hvm_info->high_mem_pgend << PAGE_SHIFT) -
                     memory_map.map[i].addr;
         memory_map.map[i].type = E820_RAM;
+        memory_map.nr_map++;
     }
 }
 
-- 
2.1.4

